solar-system.rb

```diff
@@ -0,0 +1,200 @@
+    age = age.to_i.to_s
+    age = age.reverse.scan(/\d{3}|.+/).join(",").reverse
+
+    # Calculate the closest neighboring planets
+    # Collect names and distances of planets
+    distance_and_name = []
+    @solar_system.length.times do |i|
+      planet_info = {}
+      planet_info[:name] = @solar_system[i].name
+      planet_info[:distance] = @solar_system[i].distance_from_the_sun
+      distance_and_name << planet_info
+    end
+    # Get only the distances and sort them
+    planet_distances = []
+    @solar_system.each do |planet|
+      planet_distances << planet.distance_from_the_sun
+    end
+    ordered_planets = planet_distances.sort
+    number_index = ordered_planets.index(selected.distance_from_the_sun).to_i
+
+    # Find the select planet's distance neighbors (2)
+    if (number_index+1) == @solar_system.length
+      neighbor1_distance = 0
+      neighbor1_difference = 0
+    else
+      neighbor1_distance = ordered_planets[(number_index+1)]
+      neighbor1_difference = (selected.distance_from_the_sun - neighbor1_distance).abs.round(2)
+    end
+    if (number_index-1) < 0
+      neighbor2_distance = 0
+      neighbor2_difference = 0
+    else
+      neighbor2_distance = ordered_planets[(number_index-1)]
+      neighbor2_difference = (selected.distance_from_the_sun - neighbor2_distance).abs.round(2)
+    end
+    neighbor1_name = ""
+    neighbor2_name = ""
+    distance_and_name.each do |planet|
+      if neighbor1_distance == planet[:distance]
+        neighbor1_name << planet[:name]
+      end
+      if neighbor2_distance == planet[:distance]
+        neighbor2_name << planet[:name]
+      end
+    end
+    if neighbor1_distance == 0 && neighbor2_distance == 0
+      true_neighbor = "not to be found"
+      true_distance = "both infinity and negative infinity"
+    elsif neighbor1_distance == 0 && neighbor2_distance > 0
+      true_neighbor = neighbor2_name
+      true_distance = neighbor2_difference
+    elsif neighbor2_distance == 0 && neighbor1_distance > 0
+      true_neighbor = neighbor1_name
+      true_distance = neighbor1_difference
+    else
+      true_neighbor = "#{neighbor2_name} and #{neighbor1_name}"
+      true_distance = "#{neighbor2_difference} and #{neighbor1_difference}"
+    end
+
+    # Text summary shown to user
+    text = "\nThe planet \[#{selected.name}\] is #{selected.distance_from_the_sun} Astronomical Units from the sun.
+    \rThe closest neighboring planet(s) is/are #{true_neighbor}, which is/are #{true_distance} AU away.
+    \rIt orbits the sun once every #{selected.rotation_time} Earth months, or #{(selected.rotation_time/12.0).round(2)} Earth years.
+    \rSince our sun is 4.5 billion Earth years old, this planet's age is #{age}!
+    \r#{selected.name} also enjoys #{selected.moon_count} moon(s) orbitin' around it.\n"
+    return text
+  end
+end
+
+class Planet
+  attr_accessor :name, :distance_from_the_sun, :rotation_time, :moon_count
+  # Get attributes for Planet
```

Comment:
I would like to see an instance method for `Planet` that returns a string giving details about it. There are a couple places in your `SolarSystem` class that this would be useful.